backend/app/Main.hs

```diff
@@ -15,6 +18,12 @@ import Edna.Web.Server (edna)
 
 main :: IO ()
 main = withUtf8 $ do
+  (_, Just hout, _, h) <-  createProcess (proc "poetry" ["run", "python", "analysis" </> "ic50.py"])
+    { cwd = Just $ ".." </> "analysis", std_out = CreatePipe}
```

Comment:
This doesn't look right. Running `poetry` will require network access on the host, which is not guaranteed (IIUC the intended use-case is to run Edna in an isolated environment). Also, `..` means the call depends on CWD and the fact that `analysis` dir is present in the parent directory, which will require some flimsy hacks to work in production.

```suggestion
  analysisDir <- fromMaybe (".." </> "analysis") (lookupEnv "ANALYSIS_DIR")
  (_, Just hout, _, h) <-  createProcess (proc "python3" ["analysis" </> "ic50.py"])
    { cwd = Just analysisDir, std_out = CreatePipe}
```

For development, you can just run the Haskell process inside `poetry shell`; In production, we'll run the process inside the environment generated by `poetry2nix`, and with `ANALYSIS_DIR` pointing to the correct place.